internal/ingest/ingest.go

```diff
@@ -0,0 +1,102 @@
+package ingest
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/stellar/go/ingest/ledgerbackend"
+	"github.com/stellar/go/network"
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/wallet-backend/internal/data"
+	"github.com/stellar/wallet-backend/internal/db"
+	"github.com/stellar/wallet-backend/internal/services"
+)
+
+type Configs struct {
+	DatabaseURL          string
+	NetworkPassphrase    string
+	CaptiveCoreBinPath   string
+	CaptiveCoreConfigDir string
+	LedgerCursorName     string
+	StartLedger          int
+	EndLedger            int
+}
+
+func Ingest(cfg Configs) error {
+	ctx := context.Background()
+
+	manager, err := setupDeps(cfg)
+	if err != nil {
+		log.Ctx(ctx).Fatalf("Error setting up dependencies for ingest: %v", err)
+	}
+
+	if err = manager.Run(ctx, uint32(cfg.StartLedger), uint32(cfg.EndLedger)); err != nil {
+		log.Ctx(ctx).Fatalf("Running ingest from %d to %d: %v", cfg.StartLedger, cfg.EndLedger, err)
+	}
+
+	return nil
+}
+
+func setupDeps(cfg Configs) (*services.IngestManager, error) {
+	// Open DB connection pool
+	dbConnectionPool, err := db.OpenDBConnectionPool(cfg.DatabaseURL)
+	if err != nil {
+		return nil, fmt.Errorf("error connecting to the database: %w", err)
+	}
+	models, err := data.NewModels(dbConnectionPool)
+	if err != nil {
+		return nil, fmt.Errorf("error creating models for Serve: %w", err)
+	}
+
+	// Setup Captive Core backend
+	captiveCoreConfig, err := getCaptiveCoreConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("getting captive core config: %w", err)
+	}
+	ledgerBackend, err := ledgerbackend.NewCaptive(captiveCoreConfig)
+	if err != nil {
+		return nil, fmt.Errorf("creating captive core backend: %w", err)
+	}
+
+	return &services.IngestManager{
+		NetworkPassphrase: cfg.NetworkPassphrase,
+		LedgerCursorName:  cfg.LedgerCursorName,
+		LedgerBackend:     ledgerBackend,
+		PaymentModel:      models.Payments,
+	}, nil
+}
+
+func getCaptiveCoreConfig(cfg Configs) (ledgerbackend.CaptiveCoreConfig, error) {
```

Comment:
I'm missing tests for these functions, I think at least we should validate if the values are being passed correctly.

Reply:
👍 [65edfa1](https://github.com/stellar/wallet-backend/pull/14/commits/65edfa1990c3966e6271b831f3738c4a2445dadd)